File: test_struct/struct.cpp
#include <stdio.h>
#include "struct.h"
struct_t func(struct_t a, int *ib) {
  a.a[0] = 789;
  a.b = 23.45;
  a.str[3] = 't';
  a.st->ss1 = 18;
  a.st->ss2[2] = 2;
  *ib = 10;
  return a;
}
struct_t *funcp(struct_t *a, int *ib) {
  a->a[0] = 789;
  a->b = 23.45;
  a->str[3] = 't';
  a->st->ss1 = 18;
  a->st->ss2[2] = 2;
  *ib = 10;
  return a;
}

File: test_struct/struct.js
ga = require("../lib/device/build/Debug/galileo")

obj = {
a: 15,
b: 21.36,
str: "hello",
st: {
  ss1: 2,
  ss2: 3,
}}

console.log(obj)
ga.func(obj, 5 )
console.log(obj)
ga.funcp(obj, 5 )
console.log(obj)

